### ISSUE TITLE
Discover Version Number Taks fails on RHEL

### ISSUE THREAD

**Issue by Yojimbo108** [New-Edge-Engineering/ansible-rundeck#27]
I get the below error running this on CentOS 7

```
TASK [neel.rundeck : discover version number] **********************************
fatal: [rundeck.aem.bt.com]: FAILED! => {"changed": true, "cmd": "dpkg -s rundeck | grep -i version", "delta": "0:00:00.002894", "end": "2017-03-24 11:17:55.840221", "failed": true, "rc": 1, "start": "2017-03-24 11:17:55.837327", "stderr": "/bin/sh: dpkg: command not found", "stdout": "", "stdout_lines": [], "warnings": []}
```

I think this is down to the below section in **main.yml**

```
# install rundeck
- include: install_debian.yml
  when: ansible_os_family == 'Debian'

- include: install_redhat.yml
  when: ansible_os_family == 'RedHat'

- name: discover version number
  shell: dpkg -s rundeck | grep -i version
  register: rundeck_targeting_version
```

Looks like you need to push that **discover version number** task up into **install_debian.yml** & **install_redhat.yml** and use something like `rpm -qi rundeck` on Redhat instead of dpkg.

Thanks

**Comment by mat-green** [comment]
Hello,

Thank you for this and I will do as you suggest.

M

**Comment by Yojimbo108** [comment]
Cool, I'm not sure if you want to use `rpm -qi` as it throws an ansible warning:

```
TASK [neel.rundeck : Redhat | discover version number] *************************
ok: [rundeck.aem.bt.com]
 [WARNING]: Consider using yum, dnf or zypper module rather than running rpm
```

 so I'm not sure if there's something better to use!

Thanks

**Comment by mat-green** [comment]
I am considering this fixed.

**Comment by Russell-IO** [comment]
@mat-green I was having this bug too with tag `17.03.16` (just switched the rundeck base image from Ubuntu 16.04 => Centos 7.3, however, after checking out `17.04.17` it works as expected

**Comment by mat-green** [comment]
Yes, this was fixed in 17.04.17.
